.github/CHANGELOG.md

```diff
@@ -439,93 +447,249 @@ K_test = qml.kernels.kernel_matrix(X_train, X_test, kernel)
-  This can be turned off to save memory with the Torch and TensorFlow interfaces by passing
-  `adjoint_cache=False` during QNode creation.
+* The ``argnum`` keyword argument can now be specified for a QNode to define a
+  subset of trainable parameters used to estimate the Jacobian.
+  [(#1371)](https://github.com/PennyLaneAI/pennylane/pull/1371)
+
+  For example, consider two trainable parameters and a quantum function:
+
+  ```python
+  dev = qml.device("default.qubit", wires=2)
+
+  x = np.array(0.543, requires_grad=True)
+  y = np.array(-0.654, requires_grad=True)
+
+  def circuit(x,y):
+      qml.RX(x, wires=[0])
+      qml.RY(y, wires=[1])
+      qml.CNOT(wires=[0, 1])
+      return qml.expval(qml.PauliZ(0) @ qml.PauliX(1))
+  ```
+
+  When computing the gradient of the QNode, we can specify the trainable
+  parameters to consider by passing the ``argnum`` keyword argument:
+
+  ```pycon
+  >>> qnode1 = qml.QNode(circuit, dev, diff_method="parameter-shift", argnum=[0,1])
+  >>> print(qml.grad(qnode1)(x,y))
+  (array(0.31434679), array(0.67949903))
+  ```
+
+  Specifying a proper subset of the trainable parameters will estimate the
+  Jacobian:
+
+  ```pycon
+  >>> qnode2 = qml.QNode(circuit, dev, diff_method="parameter-shift", argnum=[0])
+  >>> print(qml.grad(qnode2)(x,y))
+  (array(0.31434679), array(0.))
+  ```
+
+* Allows creating differentiable observables that return custom objects such
+  that the observable is supported by devices.
+  [(1291)](https://github.com/PennyLaneAI/pennylane/pull/1291)
+
+  As an example, first we define `NewObservable` class:
+
+  ```python
+  from pennylane.devices import DefaultQubit
+
+  class NewObservable(qml.operation.Observable):
+      """NewObservable"""
+
+      num_wires = qml.operation.AnyWires
+      num_params = 0
+      par_domain = None
+
+      def diagonalizing_gates(self):
+          """Diagonalizing gates"""
+          return []
+  ```
+
+  Once we have this new observable class, we define a `SpecialObject` class
+  that can be used to encode data in an observable and a new device that supports
+  our new observable and returns a `SpecialObject` as the expectation value:
+
+  ```python
+  class SpecialObject:
+
+      def __init__(self, val):
+          self.val = val
+
+      def __mul__(self, other):
+          new = SpecialObject(self.val)
+          new *= other
+          return new
+
+      def __imul__(self, other):
+          self.val *= other
+          return self
+
+      def __rmul__(self, other):
+          return self * other
+
+      def __iadd__(self, other):
+          self.val += other.val if isinstance(other, self.__class__) else other
+          return self
+
+      def __add__(self, other):
+          new = SpecialObject(self.val)
+          new += other.val if isinstance(other, self.__class__) else other
+          return new
+
+      def __radd__(self, other):
+          return self + other
```

Comment:
This example is great @antalszava!! Will be really helpful to have.

My only question is, do you think it's possible to reduce the code size of the `SpecialObject` class codeblock? Unfortunately, I'm not sure if that's possible without using pseudocode:

```python
class SpecialObject:
    """Container class for our observable result.
    The special methods __mul__, __imul__, __add__, __iadd__,
    __radd__ should be defined."""

    
    def __init__(self, val):
        self.val = val

    def __mul__(self, other):
        new = SpecialObject(self.val)
        new *= other
        return new

    ...
```

Reply:
Unfortunately, it seems not to be the case. The arithmetic defs are required so that the object is differentiable. Otherwise we are facing errors like:

```
RecursionError: maximum recursion depth exceeded
```
```
TypeError: unsupported operand type(s) for *: 'SpecialObject' and 'float'
```